scripts/windows/install-wsl.ps1

```diff
@@ -0,0 +1,48 @@
+# //////////////////////////////////////////////////////////////////////
+# install-wsl.ps1
+# See https://docs.microsoft.com/en-us/windows/wsl/install-win10 for the official story behind this code
+
+Param([ValidateSet("EnableWSL-01", "EnableVMPlatform-02", "InstallLinuxUpdatePackage-03")]$Step = "EnableWSL-01")
+
+$workDir = (Join-Path ([System.IO.Path]::GetTempPath()) rdinstall)
+New-Item -ItemType Directory -Force -Path $workDir
+
+$logFile = (Join-Path $workDir restarts.txt)
+$wslMsiFile = (Join-Path $workDir wsl_update_x64.msi)
+
+$script = $myInvocation.MyCommand.Definition
+$scriptPath = Split-Path -parent $script
+. (Join-Path $scriptpath restart-helpers.ps1)
+$sudoInstallScript = (Join-Path $scriptPath sudo-install-wsl.ps1)
+
+#Requires -RunAsAdministrator
+Clear-Any-Restart
+
+if ($Step -eq "EnableWSL-01") {
+  Write-Output "Doing Step EnableWSL-01"
+  Write-Output "Doing Step EnableWSL-01" | Out-File $logFile
+  dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart
+  Restart-Machine-With-Resume-Command $sudoInstallScript "EnableVMPlatform-02" "installation (step 2)"
+}
+
+if ($Step -eq "EnableVMPlatform-02") {
+  Write-Output "Doing Step EnableVMPlatform-02"
+  Write-Output "Doing Step EnableVMPlatform-02" | Out-File -Append $logFile
+  dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart
+  Restart-Machine-With-Resume-Command $sudoInstallScript "InstallLinuxUpdatePackage-03" "installation (step 3)"
+}
+
+if ($Step -eq "InstallLinuxUpdatePackage-03") {
+  Write-Output "Doing Step InstallLinuxUpdatePackage-03"
+  Write-Output "Doing Step InstallLinuxUpdatePackage-03" | Out-File -Append $logFile
+  Invoke-WebRequest -Uri https://wslstorestorage.blob.core.windows.net/wslblob/wsl_update_x64.msi -OutFile $wslMsiFile
+  msiexec /norestart /i$wslMsiFile
```

Comment:
Maybe add `/quiet` or `/passive` so the user doesn't need to click on things?

Reply:
I wanted the full visual experience in case something goes wrong.

Reply:
In that case `/passive` is probably the right option: it shows UI, but doesn't require the user to do anything to _start_ it (but should still display any errors).